### PR TITLE
WIP: Bundling restructure

### PR DESCRIPTION
**Pull request by michaelbausor**
Companion PR to https://github.com/googleapis/gax-java/pull/214

### CODE REVIEW

File: src/main/proto/com/google/api/codegen/config.proto [hunk 1]
```diff
@@ -397,6 +397,19 @@ message BundlingSettingsProto {
...
   uint32 request_byte_limit = 5;
+
+  uint32 flow_control_element_limit = 6;
+
+  uint32 flow_control_byte_limit = 7;
+
+  FlowControlLimitExceededBehavior flow_control_limit_exceeded_behavior = 8;
+}
+
+enum FlowControlLimitExceededBehavior {
```
Comment (pongad) [inline src/main/proto/com/google/api/codegen/config.proto]: Can we name this `FlowControlLimitExceededBehaviorProto`? Most types in this file has "Proto" suffix to distinguish it from similar view classes.

File: src/test/java/com/google/api/codegen/testdata/java_settings_library.baseline [hunk 2]
```diff
@@ -921,6 +990,14 @@ public class LibraryServiceSettings extends ClientSettings {
           .setRetryableCodes(RETRYABLE_CODE_DEFINITIONS.get("idempotent"))
           .setRetrySettingsBuilder(RETRY_PARAM_DEFINITIONS.get("default"));
 
+      builder.addCommentsSettings().getBundlingSettingsBuilder()
+          .setElementCountThreshold(0)
+          .setRequestByteThreshold(0)
+          .setDelayThreshold(Duration.millis(0))
```
Comment (pongad) [inline src/test/java/com/google/api/codegen/testdata/java_settings_library.baseline]: IIUC, it is invalid to set these values to zero. They should either be unset or set to positive values.

File: src/main/java/com/google/api/codegen/transformer/BundlingTransformer.java [hunk 3]
```diff
@@ -75,10 +82,32 @@ public BundlingConfigView generateBundlingConfig(MethodTransformerContext contex
...
+            .toString());
 
     return bundlingConfigView.build();
   }
 
+  private static FlowControlLimitBehavior getFlowControlLimitBehavior(
+      FlowControlLimitExceededBehavior behavior) {
+    switch (behavior) {
+      case BLOCK:
+        return FlowControlLimitBehavior.Block;
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/BundlingTransformer.java]: It seems a little silly to use an enum just to convert another constant/enum into a String. Maybe just return a String?

File: src/main/resources/com/google/api/codegen/java/settings.snip [hunk 4]
```diff
@@ -316,33 +316,35 @@
...
-              elements.addAll(request.{@desc.bundledFieldGetFunction}());
-            }
-
-            {@desc.requestTypeName} bundleRequest =
-                {@desc.requestTypeName}.newBuilder()
-                    {@fieldCopyCalls(desc.discriminatorFieldCopies)}
-                    .{@desc.bundledFieldSetFunction}(elements)
-                    .build();
-            return bundleRequest;
+          public BundlingDescriptor.RequestBuilder<{@desc.requestTypeName}> getRequestBuilder() {
```
Comment (garrettjonesgoogle) [inline src/main/resources/com/google/api/codegen/java/settings.snip]: I guess this is more of a comment on the GAX change, but I'd prefer not to use inner classes when code outside of the package has to extend/implement it.

**Comment by michaelbausor** [comment]
This PR is replaced by https://github.com/googleapis/toolkit/pull/1095

**Pull request closed without merging**